### ISSUE TITLE
More metrics for regression tasks and forecasting

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/ignite#307]
Following the discussion in #305 
We propose to implement the following list of  metrics

D1 = magnitude of error
- Mean Error
- Mean Normalized Bias
- Fractional Bias

D2 = absolute error
- Median Absolute Error
- Mean Absolute Relative Error / Mean Absolute Percentage Error
- Median Absolute Percentage Error
- Median Relative Absolute Error
- Geometric Mean Absolute Error
- Geometric Mean Relative Absolute Error
- Wave Hedges Distance
- Fractional absolute error
- Canberra Metric

based on the paper: 
https://arxiv.org/pdf/1809.03006.pdf
See appendix 2, page 30.

What do you think ?

Implementation to put into [contrib](https://github.com/pytorch/ignite/tree/master/ignite/contrib/metrics)

**Comment by anmolsjoshi** [comment]
@vfdev-5 I was interested in working on some of these metrics. Is that possible?

**Comment by vfdev-5** [comment]
@anmolsjoshi yes, I can implement one or several metrics from the list and send a PR as always :)

**Comment by zasdfgbnm** [comment]
Would there be memory issue with `Median XXX Error`? It requires recording all histories.

**Comment by vfdev-5** [comment]
@zasdfgbnm yes it can potentially lead to a memory issue if we implement it by storing everything. 
Today we already have `EpochMetric` that stores all predictions and ground-truths. BTW it worse to note that in the docs of `EpochMetric`.

**Comment by weihuangxu** [comment]
@vfdev-5 @anmolsjoshi @zasdfgbnm I think metrics are so cool. I would like to help with this issue. May I know what metrics you have been working on right now? I can start with some of the rest metrics.

**Comment by vfdev-5** [comment]
@weihuangxu I would say you can start with any metric you like (for instance avoid taking one with median) and send a PR. At the moment, there is no PRs, so they are all available :)

**Comment by anmolsjoshi** [comment]
@vfdev-5 @jasonkriss is there a reason this is closed? not all metrics from this issue have been added to ignite.

**Comment by anmolsjoshi** [comment]
@vfdev-5 is it okay to work on the metrics that save the entire history of training i.e. metrics that require calculation of median and average of the ground truth? as long as appropriate warnings are included in the docstring?
